FlowCrypt/Functionality/Services/Key Services/PassPhraseStorage.swift

```diff
@@ -8,41 +8,63 @@
 
 import UIKit
 
+// MARK: - Data Object
+struct PassPhrase: Codable, Hashable, Equatable {
+    let value: String
+    let longid: String
+
+    init(value: String, longid: String) {
+        self.value = value
+        self.longid = longid
+    }
+}
+
+// MARK: - Encrypted
+protocol EncryptedPassPhraseStorage {
+    func save(passPhrase: PassPhraseObject)
+    func update(passPhrase: PassPhraseObject)
+    func remove(passPhrase: PassPhraseObject)
+
+    func getPassPhrases() -> [PassPhraseObject]
+}
+
+// MARK: - In memory
+protocol InMemoryPassPhraseStorageType {
+    func save(passPhrase: InMemoryPassPhrase)
+    func update(passPhrase: InMemoryPassPhrase)
+    func remove(passPhrase: InMemoryPassPhrase)
+
+    func getPassPhrases() -> [InMemoryPassPhrase]
+}
```

Comment:
Since these protocols are the same (except `PassPhraseObject` vs `[InMemoryPassPhrase]`, could we unite them? And have only one type of `PassPhrase` returned?

Reply:
Also, there could be just one `PassPhraseProviderType` that covers both types of storages. Then I feel we are making actual use of the Type vs implementation distinction. That would be best.

Reply:
Actually it was done for a purpose.
Don't want to spread Realm data objects all over the place.
App layer communicates only with app layer data objects (PassPhrase) and storages with their internal objects. 
So in case smth changed (for instance we will switch to CoreDarta or any other storage) app will not be dependent on this changes.( _Besides that `InMemmoryPassPhrase` contains additionally date._)
This is common for the whole app, and actually helped us a lot when we was "moving" from "MailCore" to "GmailApi"

I will think, maybe to use `associatedObject` in protocols, to have only one, but I assume it can a bit problematic. Will check this

Reply:
Agree with not spreading realm objects everywhere. We could have a single `PassPhrase` object with a nilable expiration date on it, and then have a single protocol. But I don't insist, it's an implementation detail.

Reply:
As you suggest left only `PassPhrase` data object with optional `date`